gtsfm/data_association/point3d_initializer.py

```diff
@@ -291,17 +303,19 @@ def extract_measurements(self, track: SfmTrack2d) -> Tuple[CameraSetCal3Bundler,
         track_cameras = CameraSetCal3Bundler()
         track_measurements = Point2Vector()  # vector of 2d points
 
+        n_meas = 0
         for i, uv in track.measurements:
 
             # check for unestimated cameras
             if self.track_camera_dict.get(i) is not None:
                 track_cameras.append(self.track_camera_dict.get(i))
                 track_measurements.append(uv)
+                n_meas += 1
             else:
                 logger.warning("Unestimated cameras found at index %d. Skipping them.", i)
 
-        # triangulation is impossible with <2 measurements
-        if len(track_cameras) < 2 or len(track_measurements) < 2:
+        # triangulation is underconstrained with <2 measurements
```

Comment:
i think just len(track_cameras) is sufficient for the check, without adding the `n_meas` variable.